### PR TITLE
feat: [CCM-7515]: anomalies UAT fixes

### PR DESCRIPTION
**Pull request by jyotiarora-harness**
##### Summary:

Required fixes:
Apply Anomaly Filters [ Not Working for AWS ]
Disable Apply Anomaly Filters when there are 2 or more anomlaies on a single day
Empty state for anomalies, Error State
Overview cards not working as expected [ Error States ]
Cloud Providers filter not working [Hide]
For some of the anomalous spend, increased percentage trend is not displayed. -> N/A State
Error message for “Microsoft teams” notification method.
Order By API is failing, Show icon to show sortability
Stroke size [ increase as per mocks ]
Since x days -> x days Ago
Add Tooltip anchors
This is false anomaly -> This is a false anomaly

##### Jira Links:

https://harness.atlassian.net/browse/CCM-7515

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-ce/components/AnomaliesListView/AnomaliesListView.tsx [hunk 1]
```diff
@@ -353,12 +391,50 @@ const AnomaliesListGridView: React.FC<ListProps> = ({ listData, sortByObj, setSo
...
+        </Text>
+      </Container>
+    )
+  }
+
   /* istanbul ignore next */
   if (!listData || !listData.length) {
-    return null
+    return (
+      <Container className={css.noResultsContainer}>
```
Comment (rishik30) [inline src/modules/75-ce/components/AnomaliesListView/AnomaliesListView.tsx:428]: Too much repetition here. Can we make it a single component and pass props based on the condition?

File: src/modules/75-ce/components/AnomaliesSummary/AnomaliesSummary.tsx [hunk 2]
```diff
@@ -70,6 +77,20 @@ const AnomaliesSummary: React.FC<AnomaliesOverviewProps> = ({
 
   const statusWiseChartData = useMemo(() => transformCloudCost(statusWiseData, allDefaultProviders), [statusWiseData])
 
+  if (isAnomaliesSummaryError) {
+    return (
+      <Container className={cx(css.noResultsContainer, css.summaryError)}>
```
Comment (rishik30) [inline src/modules/75-ce/components/AnomaliesSummary/AnomaliesSummary.tsx:82]: Too much repetition here. Can we make it a single component and pass props based on the condition?

File: src/modules/75-ce/pages/anomalies-overview/AnomaliesOverviewPage.tsx [hunk 3]
```diff
@@ -269,12 +277,15 @@ const AnomaliesOverviewPage: React.FC = () => {
             cloudProvidersWiseData={cloudProvidersWiseData}
             statusWiseData={statusWiseData}
             allDefaultProviders={(ccmData?.ccmMetaData || {}) as CcmMetaData}
+            isAnomaliesSummaryError={isAnomaliesSummaryError ? Boolean(isAnomaliesSummaryError) : null}
```
Comment (rishik30) [inline src/modules/75-ce/pages/anomalies-overview/AnomaliesOverviewPage.tsx:280]: simply `Boolean(isAnomaliesSummaryError)` would also do the job here

File: src/modules/75-ce/pages/anomalies-overview/AnomaliesOverviewPage.tsx [hunk 4]
```diff
@@ -269,12 +277,15 @@ const AnomaliesOverviewPage: React.FC = () => {
...
           />
           <AnomaliesListGridView
+            searchText={searchText}
             timeRange={timeRange}
             listData={listData}
             sortByObj={sortByObj}
             setSortByObj={setSortByObj}
+            isAnomaliesListError={isAnomaliesListError ? Boolean(isAnomaliesListError) : null}
```
Comment (rishik30) [inline src/modules/75-ce/pages/anomalies-overview/AnomaliesOverviewPage.tsx:288]: simply Boolean(isAnomaliesSummaryError) would also do the job here

**Review by rishik30 — APPROVED** [review]

**Review by rajarshee-harness — APPROVED** [review]

**Pull request merged**